README
NAME

    dbcritic - Critique a database schema for best practices

VERSION

    version 0.021

USAGE

        dbcritic --help
        dbcritic --dsn dbi:Oracle:HR --username scott --password tiger
        dbcritic --class_name My::Schema --dsn dbi:mysql:database=db --username perl --password pass

DESCRIPTION

    This is the command line interface to App::DBCritic, a utility for
    scanning a database schema for violations of best practices.

CONFIGURATION

    All configuration is done via the command line options described by
    dbcritic --help.

SUPPORT

 Perldoc

    You can find documentation for this module with the perldoc command.

      perldoc bin::dbcritic

 Websites

    The following websites have more information about this module, and may
    be of help to you. As always, in addition to those websites please use
    your favorite search engine to discover more resources.

      * Search CPAN

      The default CPAN search engine, useful to view POD in HTML format.

      http://search.cpan.org/dist/App-DBCritic

      * AnnoCPAN

      The AnnoCPAN is a website that allows community annotations of Perl
      module documentation.

      http://annocpan.org/dist/App-DBCritic

      * CPAN Ratings

      The CPAN Ratings is a website that allows community ratings and
      reviews of Perl modules.

      http://cpanratings.perl.org/d/App-DBCritic

      * CPANTS

      The CPANTS is a website that analyzes the Kwalitee ( code metrics )
      of a distribution.

      http://cpants.cpanauthors.org/dist/App-DBCritic

      * CPAN Testers

      The CPAN Testers is a network of smokers who run automated tests on
      uploaded CPAN distributions.

      http://www.cpantesters.org/distro/A/App-DBCritic

      * CPAN Testers Matrix

      The CPAN Testers Matrix is a website that provides a visual overview
      of the test results for a distribution on various Perls/platforms.

      http://matrix.cpantesters.org/?dist=App-DBCritic

      * CPAN Testers Dependencies

      The CPAN Testers Dependencies is a website that shows a chart of the
      test results of all dependencies for a distribution.

      http://deps.cpantesters.org/?module=bin::dbcritic

 Bugs / Feature Requests

    Please report any bugs or feature requests through the web interface at
    https://github.com/mjgardner/dbcritic/issues. You will be automatically
    notified of any progress on the request by the system.

 Source Code

    The code is open to the world, and available for you to hack on. Please
    feel free to browse it and play with it, or whatever. If you want to
    contribute patches, please send me a diff or prod me to pull from your
    repository :)

    https://github.com/mjgardner/dbcritic

      git clone git://github.com/mjgardner/dbcritic.git

AUTHOR

    Mark Gardner <mjgardner@cpan.org>

COPYRIGHT AND LICENSE

    This software is copyright (c) 2014 by Mark Gardner.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for App-DBCritic

{{$NEXT}}
 - remove CPAN Ratings link from docs since it's read-only now
 - change search.cpan link to MetaCPAN in docs
0.023     2019-09-03 16:45:34-05:00 America/Chicago
 - bulleted Changes log to get better formatting in MetaCPAN
 - add repository links to metadata
 - inline POD to encourage keeping it up to date with the code
 - remove links to defunct AnnoCPAN
 - GH#12 add CPANFile
0.022     2019-09-03 14:24:05-05:00 America/Chicago
 - require perl v5.14 for development
 - fill out more authoring and development dependencies metadata
 - remove perlbrew release tests
 - add release tests for changelog and features for perl v5.12
 - escape possible infinite recursion if neither DSN nor schema are specified
0.021     2014-12-18 04:52:22+00:00 UTC (TRIAL RELEASE)
 - #9 remove experimental smartmatch operators (HAARG++, SREZIC++)
 - #11 fix POD encoding (SROMANV++)
0.020     2011-09-15 15:32:37 America/New_York
 - rename to App-DBCritic
0.014     2011-08-30 16:13:33 America/New_York
 - bump up required perl to 5.10.1 to deal with Moo & Sub::Quote issues
 - some minor POD, test and code readability fixes
0.013     2011-08-16 10:34:11 America/New_York
 - use DBICSC::PolicyType roles for applies_to attributes in policies,
    - should make writing plugins for multiple types of objects as easy
      as composing multiple roles
 - explicit version requirements for *both* DBIC and DBIC::Schema::Loader
 - experimenting with inlining some defaults with Sub::Quote
0.012     2011-08-06 18:57:20 America/New_York
 - bump DBIC::Schema::Loader requirement to 0.07007
 - add DBICSC::Policy::DuplicateRelationships (requested by frew++)
0.011     2011-08-06 13:49:04 America/New_York
 - require more recent version of DBIx::Class
 - fix synopsis (reported by frew++)
 - fill in DBICSC::Policy::NullableTextColumn pod with data types checked
0.010     2011-07-29 16:44:18 America/New_York
 - converted from Moose to Moo object framework (requested by mst++)
0.005     2011-07-29 13:04:52 America/New_York
 - dsn is now a required attribute, defaults to schema object's storage.
 - cache stringification of Violation objects
0.004     2011-07-28 18:29:40 America/New_York
 - initial release


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.029.
use strict;
use warnings;

use 5.012;

use ExtUtils::MakeMaker;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Critique a database schema for best practices",
  "AUTHOR" => "Mark Gardner <mjgardner\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "App-DBCritic",
  "EXE_FILES" => [
    "bin/dbcritic"
  ],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.012",
  "NAME" => "App::DBCritic",
  "PREREQ_PM" => {
    "Algorithm::Combinatorics" => 0,
    "Carp" => 0,
    "Const::Fast" => 0,
    "DBI" => 0,
    "DBIx::Class" => "0.08125",
    "DBIx::Class::Schema::Loader" => "0.07007",
    "Data::Compare" => 0,
    "Devel::Symdump" => 0,
    "English" => 0,
    "Getopt::Long::Descriptive" => 0,
    "List::MoreUtils" => 0,
    "List::Util" => "1.33",
    "Modern::Perl" => 0,
    "Module::Pluggable" => 0,
    "Moo" => 0,
    "Moo::Role" => 0,
    "Scalar::Util" => 0,
    "Sub::Quote" => 0,
    "namespace::autoclean" => 0,
    "overload" => 0,
    "strict" => 0,
    "utf8" => 0
  },
  "TEST_REQUIRES" => {
    "DBICx::TestDatabase" => 0,
    "DBIx::Class::Core" => 0,
    "DBIx::Class::Schema" => 0,
    "File::Spec" => 0,
    "FindBin" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Path::Class" => 0,
    "Test::More" => "0.94",
    "Test::Most" => 0,
    "Test::Script::Run" => 0,
    "base" => 0,
    "local::lib" => 0,
    "warnings" => 0
  },
  "VERSION" => "0.021",
  "test" => {
    "TESTS" => "t/*.t t/policy/*.t"
  }
);


my %FallbackPrereqs = (
  "Algorithm::Combinatorics" => 0,
  "Carp" => 0,
  "Const::Fast" => 0,
  "DBI" => 0,
  "DBICx::TestDatabase" => 0,
  "DBIx::Class" => "0.08125",
  "DBIx::Class::Core" => 0,
  "DBIx::Class::Schema" => 0,
  "DBIx::Class::Schema::Loader" => "0.07007",
  "Data::Compare" => 0,
  "Devel::Symdump" => 0,
  "English" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "FindBin" => 0,
  "Getopt::Long::Descriptive" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "List::MoreUtils" => 0,
  "List::Util" => "1.33",
  "Modern::Perl" => 0,
  "Module::Pluggable" => 0,
  "Moo" => 0,
  "Moo::Role" => 0,
  "Path::Class" => 0,
  "Scalar::Util" => 0,
  "Sub::Quote" => 0,
  "Test::More" => "0.94",
  "Test::Most" => 0,
  "Test::Script::Run" => 0,
  "base" => 0,
  "local::lib" => 0,
  "namespace::autoclean" => 0,
  "overload" => 0,
  "strict" => 0,
  "utf8" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



